### DISCUSSION TITLE
Regression Testing Expectations

### DISCUSSION THREAD

**Discussion by JacksonBurns** [ReactionMechanismGenerator/RMG-Py#2476] [General]
## Regression Testing

As of right now, the regression tests are failing by default. See this comment for both the failure, and an initial reaction:

              > Not related to this PR but this is the first PR where the regression tests have had a fair chance to run. The tests failed because one edge reaction was different. From the failed edge comparison:

That result suggests a bug to me...one that may be quite complicated, but probably shouldn't be discussed here in this PR.

_Originally posted by @mjohnson541 in https://github.com/ReactionMechanismGenerator/RMG-Py/issues/2474#issuecomment-1601283617_

Let's discuss what suggests bugs, what should constitute a failure, and how much reproducibility we expect from RMG runs.

**Comment by mjohnson541** [comment]
When any react generates CC[CH]OO when constructing the reaction the radical decay framework should convert CC[CH]OO to CCC=O + [OH] since it matches the template for a radical decay recipe.

**Reply by JacksonBurns** [reply]
Is it possible then that, when CC[CH]OO was generated nothing was moved to the core that iteration, so the simulation stopped before the next reaction could be added?

**Reply by mjohnson541** [reply]
I don't understand, isn't the text indicating it was in the model (which I assume means the core)?

It's possible that the check only occurs when adding to the core (I don't think that's the case though), but regardless of that you would expect consistency in the reaction whether both were in the core or edge.

**Comment by JacksonBurns** [comment]
Notes from a developer meeting - the regression tests should 'pass' (just in the sense that we get a green checkmark from GitHub actions) if they successfully _execute_. The results of the actual regression comparisons should be commented onto the PR by a bot in a dropdown so that reviewers can look at the results.
